js/stdlib/ts/rpc_server.ts

```diff
@@ -26,84 +27,177 @@ const withApiKey = () => {
 };
 
 
-export const mkStdlibProxy = async (lib: any) => {
-  const makeHandle = (container: Array<any>) => (val: any) => {
-    const id      = container.length;
-    container[id] = val;
-    return id;
+const mkKont = () => {
+  // TODO consider replacing stringly-typed exceptions with structured
+  // descendants of `Error` base class
+  const COLLISION = 'Collision on continuation ID:';
+  const UNTRACKED = 'Untracked continuation ID:';
+  const collision = (i: string) => `${COLLISION} ${i}`;
+  const untracked = (i: string) => `${UNTRACKED} ${i}`;
+  const k: any    = {};
+
+  const mkWas = (m: string) => (e: Error): boolean =>
+    !!(e.message
+      .substr(0, m.length)
+      .match(`^${m}$`));
+
+  const was = {
+    collision: mkWas(COLLISION),
+    untracked: mkWas(UNTRACKED),
+  };
+
+  const raise = (e: string) => {
+    throw new Error(e);
+  };
+
+  const track = (a: any, bytes: number = 24) =>
+    new Promise<string>((res, rej) => randomBytes(bytes, (e, b) =>
```

Comment:
As mentioned in Slack, I think you should change this to `randomBytes() + "_" + id++` so there's no risk of collision, get rid of `collision` and `safely`. Also, I prefer `async` over explicitly making Promises. I am too weak of a programmer to hand Promises themselves

Reply:
How does 5debabcc6d4d76c8313a3b7c37be85b0d9a2e4d7 look?

I've left the "untracked" bits as they were because I think it's important to capture those and do something deliberate at the HTTP layer.